lib/sth_database.js

```diff
@@ -1421,18 +1509,259 @@ function storeRawData(data, callback) {
+ *  - {string} entityType The entity type associated to the updated attribute
+ *  - {object} attribute The updated attribute
+ *  - {Date} timestamp The timestamp associated to the updated raw data
+ * @param callback The callback to notify in case of error or once the new maximum value has been calculated
+ */
+function getNewMinValues(data, callback) {
+  function getNewMinValue(data, resolution, callback) {
+    data.collection.find(
+      getNewMinMaxCondition(data, resolution)
+    ).sort(
+      {
+        attrValue: -1
+      }
+    ).toArray(
+      function (err, restArray) {
+        if (callback) {
+          return process.nextTick(
+            callback.bind(null, err, (restArray && restArray.length) ? restArray[0].attrValue : null));
+        }
+      }
+    );
+  }
+  var getNewMinValue4Resolutions = {};
+  for (var i = 0; i < sthConfig.AGGREGATION_BY.length; i++) {
+    getNewMinValue4Resolutions[sthConfig.AGGREGATION_BY[i]] =
+      getNewMinValue.bind(null, data, sthConfig.AGGREGATION_BY[i]);
+  }
+  async.parallel(
+    getNewMinValue4Resolutions,
+    function (err, result) {
+      if (callback) {
+        return process.nextTick(callback.bind(null, err, result));
+      }
+    }
+  );
+}
+
+/**
+ * Returns information about the notification such as if the notification aims to insert new data,
+ *  update already existent data or if it corresponds to already registered data
+ * @param {object} data The data to be stored. It is an object including the following properties:
+ *  - {object} collection: The collection where the data should be stored in
+ *  - {date} recvTime The date the event arrived
+ *  - {string} entityId The entity id associated to the updated attribute
+ *  - {string} entityType The entity type associated to the updated attribute
+ *  - {object} attribute The updated attribute
+ * @param {Function} callback Function to call once the operation completes
+ */
+function getNotificationInfo(data, callback) {
+  var collection = data.collection,
+    recvTime = data.recvTime,
+    entityId = data.entityId,
+    entityType = data.entityType,
+    attribute = data.attribute;
+
+  var timestamp = sthHelper.getAttributeTimestamp(attribute, recvTime);
+  data.timestamp = timestamp;
+
+  var findCondition;
+  switch (DATA_MODEL) {
+    case DATA_MODELS.COLLECTIONS_PER_SERVICE_PATH:
+      findCondition = {
+        recvTime: timestamp,
+        entityId: entityId,
+        entityType: entityType,
+        attrName: attribute.name,
+        attrType: attribute.type
+      };
+      break;
+    case DATA_MODELS.COLLECTIONS_PER_ENTITY:
+      findCondition = {
+        recvTime: timestamp,
+        attrName: attribute.name,
+        attrType: attribute.type
+      };
+      break;
+    case DATA_MODELS.COLLECTIONS_PER_ATTRIBUTE:
+      findCondition = {
+        recvTime: timestamp,
+        attrType: attribute.type
+      };
+      break;
+  }
+
+  collection.findOne(
+    findCondition,
+    function(err, result) {
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-sth-comet/commit/190f1c31cc47d640070c423c4bea43ad264384cc
Huge improvements in readability in my honest opinion, thanks!!! ;) More improvements to come in a forthcoming refactoring as a result of solving https://github.com/telefonicaid/fiware-sth-comet/issues/222